src/auro-combobox.js

```diff
@@ -656,25 +662,52 @@ export class AuroCombobox extends LitElement {
   }
 
   /**
-   * Watch for slot changes and recalculate the menuoptions.
+   * Recalculate the menuoptions on deep slot changes.
    * @private
    * @returns {void}
    */
-  handleSlotChange() {
-    if (this.auroMenuReady) {
-      this.options = this.menu.querySelectorAll('auro-menuoption, [auro-menuoption]');
-      this.options.forEach((opt) => {
-        if (this.checkmark) {
-          opt.removeAttribute('nocheckmark');
-        } else {
-          opt.setAttribute('nocheckmark', '');
-        }
-      });
+  handleDeepSlotChanges() {
+    this.updateMenuOptions();
+    this.showBib();
+  }
+
+  /**
+   * Menu specific updates.
+   * @private
+   * @returns {void}
+   */
+  updateMenuOptions() {
+    if (!this.auroMenuReady) {
+      return;
     }
 
+    this.options = this.menu.querySelectorAll('auro-menuoption, [auro-menuoption]');
+    this.options.forEach((opt) => {
+      opt.toggleAttribute('nocheckmark', !this.checkmark);
+    });
     this.handleMenuOptions();
   }
 
+  /**
+   * Observe child nodes in slot for updates.
+   * @private
+   * @returns {void}
+   */
+  observeSlotChanges() {
+    const [slotContent] = this.shadowRoot.querySelector("#defaultSlot").assignedElements();
```

Comment:
```suggestion
    const [slotContent] = this.shadowRoot.querySelector(".menuWrapper slot").assignedElements();
```

Just a suggestion on another way to do this without adding to the DOM

Reply:
updated